### PR TITLE
Check tenant active status before SP retrieval

### PR DESCRIPTION
**Pull request by Avarjana**
$Subject

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/TenantManagementListener.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package org.wso2.carbon.identity.application.mgt.listener;
```
Comment (senthalan) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/TenantManagementListener.java:1]: license missing

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/TenantManagementListener.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+        try {
+            String tenantDomain = IdentityTenantUtil.getTenantDomain(tenantID);
+            ApplicationBasicInfo[] applicationBasicInfos = applicationDAO.getAllApplicationBasicInfo(tenantID);
+            for (ApplicationBasicInfo applicationBasicInfo: applicationBasicInfos) {
+                ServiceProvider sp = new ServiceProvider();
+                sp.setApplicationName(applicationBasicInfo.getApplicationName());
+                sp.setApplicationID(applicationBasicInfo.getApplicationId());
+                sp.setApplicationResourceId(applicationBasicInfo.getApplicationResourceId());
+                cacheBackedApplicationDAO.clearApplicationFromCache(sp, tenantDomain);
```
Comment (Avarjana) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/TenantManagementListener.java:34]: Redundant cache clear. Don't delete cache entry only.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java [hunk 3]
```diff
@@ -1007,6 +1007,10 @@ public static void startTenantFlow(String tenantDomain) throws IdentityApplicati
```
Comment (Avarjana) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:1007]: Make sure this is called from all the application service impl.

Comment (Avarjana) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:1007]: Fixed with [802eef3](https://github.com/wso2/carbon-identity-framework/pull/4361/commits/802eef3e68825657fbf77b4e1ae3fe726ed4bebc)

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/TenantManagementListener.java (hunk 2 above)
Comment (Avarjana) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/TenantManagementListener.java]: Fixed with [802eef3](https://github.com/wso2/carbon-identity-framework/pull/4361/commits/802eef3e68825657fbf77b4e1ae3fe726ed4bebc)

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/TenantManagementListener.java (hunk 1 above)
Comment (Avarjana) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/TenantManagementListener.java]: Fixed with [802eef3](https://github.com/wso2/carbon-identity-framework/pull/4361/commits/802eef3e68825657fbf77b4e1ae3fe726ed4bebc)

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java [hunk 4]
```diff
@@ -1007,6 +1007,10 @@ public static void startTenantFlow(String tenantDomain) throws IdentityApplicati
 
         String userId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUserId();
         int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        if (MultitenantConstants.SUPER_TENANT_ID != tenantId && !IdentityTenantUtil.getTenant(tenantId).isActive()) {
+            throw new IdentityApplicationManagementException("Error when setting tenant domain. "
```
Comment (senthalan) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:1011]: This should a client Exception. The messaging needs to change. `Error when setting tenant domain` this is wrong

Comment (Avarjana) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:1011]:
```
"throw new IdentityApplicationManagementException("Cannot start the tenant flow. "
                    + tenantDomain + " is deactivated.");
```
WDYT?

Comment (senthalan) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:1011]: yeah, and it should be IdentityApplicationManagementClinetException

Comment (Avarjana) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/4361/commits/cc2d053fa557680f301bf9a3ce064c832b5fbc1b

**Review by senthalan — DISMISSED** [review]

**Review by senthalan — APPROVED** [review]

**Pull request merged**